From dc44690985e727ac2ef5af783a495a369223ab69 Mon Sep 17 00:00:00 2001
From: Jim Meyering <meyering@redhat.com>
Date: Wed, 30 Mar 2011 08:46:22 +0200
Subject: [PATCH] tests: readlink* ("",... fails with EINVAL on newer kernels

readlink and readlinkat have typically failed with ENOENT for
the invalid, empty file name,  "".  However, with the advent
of linux-2.6.39, they fail with EINVAL.
* tests/test-areadlink.h (test_areadlink): Also accept EINVAL
when operating on the empty file name.
* tests/test-readlink.h (test_readlink): Likewise.
---
 ChangeLog              |   10 ++++++++++
 tests/test-areadlink.h |    2 +-
 tests/test-readlink.h  |    2 +-
 3 files changed, 12 insertions(+), 2 deletions(-)

diff --git a/gettext-tools/gnulib-tests/test-areadlink.h b/gettext-tools/gnulib-tests/test-areadlink.h
index c7d8324..55b41c0 100644
--- a/gettext-tools/gnulib-tests/test-areadlink.h
+++ b/gettext-tools/gnulib-tests/test-areadlink.h
@@ -36,7 +36,7 @@ test_areadlink (char * (*func) (char const *, size_t), bool print)
   ASSERT (errno == ENOENT);
   errno = 0;
   ASSERT (func ("", 1) == NULL);
-  ASSERT (errno == ENOENT);
+  ASSERT (errno == ENOENT || errno == EINVAL);
   errno = 0;
   ASSERT (func (".", 1) == NULL);
   ASSERT (errno == EINVAL);
diff --git a/gettext-tools/gnulib-tests/test-readlink.h b/gettext-tools/gnulib-tests/test-readlink.h
index 08d5662..7247fc4 100644
--- a/gettext-tools/gnulib-tests/test-readlink.h
+++ b/gettext-tools/gnulib-tests/test-readlink.h
@@ -38,7 +38,7 @@ test_readlink (ssize_t (*func) (char const *, char *, size_t), bool print)
   ASSERT (errno == ENOENT);
   errno = 0;
   ASSERT (func ("", buf, sizeof buf) == -1);
-  ASSERT (errno == ENOENT);
+  ASSERT (errno == ENOENT || errno == EINVAL);
   errno = 0;
   ASSERT (func (".", buf, sizeof buf) == -1);
   ASSERT (errno == EINVAL);
